bindings/cpp/test_array.cpp

```diff
@@ -0,0 +1,115 @@
+#include "array.h"
+#include "datapipeline.hh"
+
+#include <iostream>
+
+#include "pybind11/embed.h"
+#include "pybind11/numpy.h"
+#include "pybind11/stl.h"
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+const std::string TEST_HDF5_FILENAME = "test_npy.h5";
+const char *TEST_DATASET_NAME = "nparray";
+
+/// std api applies here, but python open(hdf5 file is not impl, use h5py to open file
+void write_array(const string &data_product, const string &component, const Array &da)
```

Comment:
What is the purpose of this? It should just call the python api's write_array function.   It looks like write_array has been reimplemented here, but using h5py.  There's no point implementing this here; just wait until we have the standard API writing HDF5 files, then wrap that.

Reply:
The PR implementing the HDF5 writing from python has been merged, fyi.